source/dagger/gameplay/ping_pong/player_scores.cpp

```diff
@@ -50,6 +50,7 @@ void PlayerScoresSystem::Run()
     if (ballOnField == 0)
     {
         // TODO: set speed to be random in both directions
-        CreatePingPongBall(s_TileSize, ColorRGBA(1, 1, 1, 1), { rand()%10 + 4,rand()%10 + 4,0 },   { 0,rand()%(s_FieldHeight / 2),0 });
+		float x_speed = rand() % 10 - 3.12;
```

Comment:
x_speed can be generated with 0, so the game will freeze, please try to fix it

Reply:
Also, `3.12` is a double and not a float (like `3.12f` is). Watch out for type sizes and implicit conversions...